examples/games/loading_screen.rs

```diff
@@ -204,21 +204,13 @@ fn update_loading_data(
         // we reset the confirmation frame count.
         loading_data.confirmation_frames_count = 0;
 
-        // Go through each asset and verify their load states.
-        // Any assets that are loaded are then added to the pop list for later removal.
-        let mut pop_list: Vec<usize> = Vec::new();
-        for (index, asset) in loading_data.loading_assets.iter().enumerate() {
+        loading_data.loading_assets.retain(|asset| {
             if let Some(state) = asset_server.get_load_states(asset) {
```

Comment:
nit: could be
```rust
asset_server
    .get_recursive_dependency_load_state(asset)
    .map_or(true, |state| !state.is_loaded())
```

Reply:
Actually, I think that would trigger a lint for `is_none_or`. :p

Reply:
It doesn't but that is even nicer :)